pkg/util/k8s/cluster.go

```diff
@@ -0,0 +1,42 @@
+package k8s
+
+import (
+	"context"
+
+	ocp_configv1 "github.com/openshift/api/config/v1"
+	metaerrors "k8s.io/apimachinery/pkg/api/meta"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IsClusterBeingUpgraded checks if the Kubernetes cluster is being upgraded.
+// Currently this only checks for OpenShift clusters.
+func IsClusterBeingUpgraded(k8sClient client.Client) (bool, error) {
+	cvList := &ocp_configv1.ClusterVersionList{}
+	err := k8sClient.List(context.TODO(), cvList, &client.ListOptions{})
+	if metaerrors.IsNoMatchError(err) {
```

Comment:
nit: when is this going to be executed? even if this is executed it is still ok to not have the if clause, and return false err.

Reply:
This error happens when the k8s api server does have the `ClusterVersion` kind registered. Which would be almost all k8s clusters except OpenShift. Not returning error to avoid logging this as a warning in the caller as we should ignore this silently in non-openshift cases.